src/bin/lint.ts

```diff
@@ -126,38 +127,57 @@ async function main(argv = process.argv) {
   // Always include README if it exists
   const markdownFiles: string[] = [];
   if (fs.existsSync('README.md')) markdownFiles.push('README.md');
-
-  // Add files from pages/, blog/, docs/ **if they exist AND contain md/mdx**
   for (const dir of ['pages', 'blog', 'docs']) {
-    if (fs.existsSync(dir)) {
-      markdownFiles.push(...utils.collectMarkdown(dir));
-    }
+    if (fs.existsSync(dir)) markdownFiles.push(...utils.collectMarkdown(dir));
   }
-
   if (markdownFiles.length === 0) {
     console.warn('Skipping Prettier: no Markdown/MDX files found.');
     return;
   }
 
   const prettierArgs = [fix ? '--write' : '--check', ...markdownFiles];
-
   console.error('Running prettier:');
-  console.error(' ' + ['prettier', ...prettierArgs].join(' '));
 
+  const require = createRequire(import.meta.url);
+  let prettierBin: string | null = null;
   try {
-    childProcess.execFileSync('prettier', prettierArgs, {
-      stdio: 'inherit',
-      windowsHide: true,
-      encoding: 'utf-8',
-      shell: platform === 'win32',
-      cwd: process.cwd(),
-    });
+    // Resolves to @matrixai/lint/node_modules/prettier/bin/prettier.cjs
+    prettierBin = require.resolve('prettier/bin/prettier.cjs');
```

Comment:
Don't use `createRequire` this is hack to allow for synchronous dynamic importing of code. If you need to resolve a path to an absolute use `import.meta.url` to get the current directory and use `path.resolve` to resolve that to an absolute path.